spacetime/tests/test_norm.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+import unittest
+import libst
+
+''' Calculate the vector norm
+    vec: vector for norm calculation
+    ord: norm order
+'''
+def norm(vec, ord = 1):
+    res = 0 
+    for ele in vec:
+        res += abs(ele) ** ord
+    return res ** (1 / ord)
+
+
+''' Calculate exact solution of the 1D advection equation
+    svr: linear wave solver
+    iter_num: how many step will be executed
+    gfun: system function
+'''
+def ExactSolution(svr, iter_num, gfun):
+    v = np.array([])
+    for e in svr.selms(odd_plane = False):
+        x = (e.xctr - iter_num * svr.dt) % svr.grid.xmax  # dealt with boundary treatment
+        if x < 2 * np.pi or x > 4 * np.pi:
+            v = np.append(v, 0)
+        else:
+            v = np.append(v, gfun(x))
+    return v
+
+class NormTC(unittest.TestCase):
+
+    @staticmethod
+    def _build_solver(resolution):
+
+        grid = libst.Grid(0, 4 * 2 * np.pi, resolution)
+        cfl = 1
+        dx = (grid.xmax - grid.xmin) / grid.ncelm
+        dt = dx * cfl 
+        svr = libst.LinearScalarSolver(grid = grid, time_increment = dt)
```

Comment:
Why don't you add the test case in `test_linear_scalar.py`?

Reply:
Oh...you are right, I forgot there is a ```test_linear_scalar.py```.
I will move norm test into ```test_linear_scalar.py```.